Add bazel release-1.12 jobs and add release-1.12 testgrid dashboards.

cjwagner opened the pull request:
We still need to update the existing testgrid release dashboards to point to the correct targets.

/cc @amwat @krzyzacy 
/area jobs


Hold this until we determine what order to merge all the release PRs.
/hold

krzyzacy commented:
the periodic variant has release-1.8 branch jobs, probably get rid of those?

it can be merged, it won't run until we have 1.12 branches anyway

krzyzacy commented:
also add a `# manual-release-bump-required` comment?

cjwagner commented:
I added the comment and deleted the 1.8 jobs.

This still need to wait to merge so that I can update the image tags once the new `kubekins-e2e` has been published.

krzyzacy commented on config/jobs/kubernetes/sig-testing/bazel-build-test.yaml:
```diff
@@ -144,7 +145,38 @@ postsubmits:
+    spec:
+      containers:
+        # TODO(cjwagner): Change to 1.12 image tag when available.
+      - image: gcr.io/k8s-testimages/kubekins-e2e:v20180730-8b7ab3104-master
```
you should be able to use the 1.12 images pushed by @amwat now

cjwagner replied:
Done.

cjwagner commented:
I think this one is ready to merge now. PTAL.
/hold cancel

krzyzacy commented:
/lgtm
